core/src/main/java/org/web3j/protocol/websocket/WebSocketService.java

```diff
@@ -98,9 +99,14 @@ public WebSocketService(WebSocketClient webSocketClient, boolean includeRawRespo
      * @throws ConnectException thrown if failed to connect to the server via WebSocket protocol
      */
     public void connect() throws ConnectException {
+        connect(null, null, null);
```

Comment:
I prefer to pass empty lambdas here and not check for null in the listener methods.